SaltSquareIO/sports-app-react-fe
# Login API

[CLICKUP_TICKET_ID](https://app.clickup.com/t/86c0g7mt4)

## Summary of change

This PR implements the login functionality. Key changes include:

1. Finalized the login flow by making a call to the `/auth/login` endpoint.
2. Upon successful login, the user is redirected to the home page.
3. Added error handling to display "Incorrect email or password." message if the login attempt fails for any reason.
4. Implemented logic to automatically clear the error message when the user starts typing again in the email or password input field. 

## QA Notes

To test this PR, follow these instructions: 

1. Attempt to log in with incorrect credentials. The "Incorrect email or password." message should appear. 
2. After typing in either the email or password field, the error message should disappear. 
3. Log in with correct credentials and ensure the user is redirected to the home page. 

## Screenshots

### Error message if the request is not successful 

<img width="511" alt="Screenshot 2024-10-24 at 12 17 04" src="https://github.com/user-attachments/assets/a9f69e4f-93bd-415a-8e77-967ea9e84429">
